### PR TITLE
Add Cypress Tests for National Delivery Address Blocking

### PR DESCRIPTION
**Pull request by andrewHEguardian**
## What does this change?

Adds Cypress tests for National Delivery address updates being disabled. A small refactoring of the check to do this is included in the PR.

## How to test

Cypress tests pass. 

## Accessibility

-   [ ] [Tested with screen reader](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#screen-reader)
-   [ ] [Navigable with keyboard](https://github.com/guardian/accessibility/blob/main/people-and-technology/02-physical.md#keyboard)
-   [ ] [Colour contrast passed](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#contrast)
-   [ ] [The change doesn't use only colour to convey meaning](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#use-of-colour)

### CODE REVIEW

File: cypress/e2e/parallel-4/deliveryAddress.cy.ts [hunk 1]
```diff
@@ -76,29 +77,38 @@ describe('Delivery address', () => {
...
+	it('Shows updated address when returning to manage subscription page', () => {
+		cy.intercept('GET', '/api/me/mma', {
 			statusCode: 200,
-			body: 'success',
-		}).as('address_update');
+			body: toMembersDataApiResponse(
+				guardianWeeklyPaidByCard(),
+				supporterPlus(),
+			),
+		}).as('mma');
```
Comment (charleycampbell) [inline cypress/e2e/parallel-4/deliveryAddress.cy.ts:112]: Is it worth adding in a test to check the edge case where someone has a home delivery and a national delivery they cannot update their address?

**Review by charleycampbell — APPROVED** [review]
Looks good to me. Just one suggestion to add in an additional test for the edge where a user has both a home delivery and national delivery product.

File: cypress/e2e/parallel-4/deliveryAddress.cy.ts (hunk 1 above)
Comment (andrewHEguardian) [reply cypress/e2e/parallel-4/deliveryAddress.cy.ts:112]: yeah that's a good shout

**Review by KentGrm — APPROVED** [review]

File: cypress/e2e/parallel-4/deliveryAddress.cy.ts (hunk 1 above)
Comment (andrewHEguardian) [reply cypress/e2e/parallel-4/deliveryAddress.cy.ts:112]: will do in separate PR
